src/main/java/com/k3ntako/HTTPServer/Request.java

```diff
@@ -0,0 +1,96 @@
+package com.k3ntako.HTTPServer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class Request implements RequestInterface {
+  private String method;
+  private String route;
+  private String protocol;
+  private HashMap<String, String> headers;
+  private String body;
+
+  public Request() {
+    this.headers = new HashMap<>();
+  }
+
+  public void parseRequest(BufferedReader bufferedReader) {
```

Comment:
I think I would lean towards having a single public `parse` method and let the guts of this class worry about whether there is a body or not. That way you don't have to know to call two methods if this is a POST rather than a GET.